src/Sarif/Writers/SarifLogger.cs

```diff
@@ -175,11 +180,15 @@ public void Dispose()
             {
                 _issueLogJsonWriter.CloseResults();
 
-                if (_run != null &&
-                    _run.Invocation != null &&
-                    _run.Invocation.StartTime != new DateTime())
+                if (_run?.Invocations != null)
                 {
-                    _run.Invocation.EndTime = DateTime.UtcNow;
+                    foreach (var invocation in _run.Invocations)
```

Comment:
This is a tough one. I don't know what the right answer is, but this isn't it :-)

The reason `invocations` is an array is that some tools perform an analysis by executing a sequence of programs (rather than just one program). Each program in the sequence needs its own invocation. And of course, each has its own start time.

Just populate the _first_start time for now.

Reply:
Done